README.md

```diff
@@ -1,2 +1,130 @@
 # mocha-appium-app-browserstack
- Mocha nodeJS Appium integration with BrowserStack
+
+This repository demonstrates how to run Appium NodeJS mocha tests on BrowserStack App Automate.
+
+## Setup
+
+### Requirements
+
+1. Node.js
+
+    - If not installed, install Node.js from [here](https://nodejs.org/en/download/)
+
+    - Ensure you have node & npm installed by running `node -v` & `npm -v`
+
+### Install the dependencies
+
+For Android tests, run the following command in project's base directory :
+
+```
+cd android
+npm install
+```
+Or,
+
+For dependencies for iOS tests, run following command in project's base directory :
+```
+cd ios
+npm i
+```
+## Getting Started
+
+Getting Started with Appium tests in NodeJS mocha   on BrowserStack couldn't be easier!
+
+### Run your first test :
+
+**1. Upoad your Android or iOS App**
+
+Upload your Android app (.apk or .aab file) or iOS app (.ipa file) to BrowserStack servers using our REST API. Here is an example cURL request :
+
+```
+curl -u "YOUR_USERNAME:YOUR_ACCESS_KEY" \
+-X POST "https://api-cloud.browserstack.com/app-automate/upload" \
+-F "file=@/path/to/apk/file"
```

Comment:
is upload via url needed ?

Reply:
You mean, dashboard upload should be here?
Rest of the sample repos suggests to use rest api to upload the app.